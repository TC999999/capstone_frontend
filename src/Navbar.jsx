import { Link, useNavigate } from "react-router-dom";
import { useContext } from "react";
import UserContext from "./UserContext";
import "./styles/Navbar.css";

const NavBar = () => {
  const navigate = useNavigate();
  const { user, logOut } = useContext(UserContext);

  const logOutAndNavigate = () => {
    logOut();
    navigate("/");
  };

  if (!user) {
    return (
      <div className="navbar">
        <Link to="/">Home</Link>
        <Link to="/register">
          <button className="auth-button">Sign Up/Login</button>
        </Link>
      </div>
    );
  }

  return (
    <div className="navbar">
      <Link to="/">Home</Link>
      <Link to="/items/search/all">Search for Items</Link>
      <Link to={`/users/${user.username}/messages`}>Messages</Link>
      {user.isAdmin && <Link to="reports">Reports</Link>}
      <span>
        <Link to={`/users/${user.username}`}>{user.username}</Link>
        <button className="auth-button" onClick={logOutAndNavigate}>
          Log Out
        </button>
      </span>
    </div>
  );
};

export default NavBar;
