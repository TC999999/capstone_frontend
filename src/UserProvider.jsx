import UserContext from "./UserContext.js";
import { useState, useEffect } from "react";
import marketAPI from "../api.js";

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  //sets context for user for entire app
  const currentUser = (user) => {
    setUser(user);
  };

  // //On initial render, get user token from local storage, gets the current user from the api, and set as context
  useEffect(() => {
    async function getToken() {
      let token = localStorage.getItem("market-token");
      if (token) {
        let userInfo = await marketAPI.getCurrentUser();
        currentUser(userInfo);
      }
    }
    getToken();
  }, []);

  //These functions and user state will be used in context
  //Set the token in local storage, retrives the current user from the api, and sets the user as context on login
  const logIn = async (username, token) => {
    localStorage.setItem("market-token", token);
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  //remove token and user from local storage and set user context to null on logout
  const logOut = () => {
    localStorage.removeItem("market-token");
    currentUser(null);
  };

  //update current user in local storage and context
  const updateUser = async (username) => {
    let userInfo = await marketAPI.getUserInfo(username);
    currentUser(userInfo);
  };

  return (
    <UserContext.Provider
      value={{ user, currentUser, logIn, logOut, updateUser }}
    >
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
