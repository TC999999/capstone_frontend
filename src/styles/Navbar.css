.navbar {
  background-color: #88aeeb;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.navbar a {
  color: white;
  margin: 5px;
  font-family: Arial, Helvetica, sans-serif;
  text-decoration-line: none;
}

.navbar span {
  margin: 5px;
}

.navbar a:hover {
  color: purple;
  background-color: aqua;
}

.navbar a:active {
  color: purple;
  background-color: white;
}

.auth-button {
  background-color: blue;
  color: white;
  border: solid 1px;
  border-radius: 5px;
  padding: 2px;
  font-weight: bold;
}

.auth-button:hover {
  background-color: aqua;
  color: purple;
}

.auth-button:active {
  background-color: purple;
  color: white;
}
