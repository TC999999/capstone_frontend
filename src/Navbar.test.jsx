import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "./UserProvider.jsx";
import Navbar from "./Navbar.jsx";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Navbar />
      </UserProvider>
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Navbar />
      </UserProvider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
