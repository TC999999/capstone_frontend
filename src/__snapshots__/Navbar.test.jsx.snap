// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="nav-box"
    >
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="nav-box"
    >
      <a
        href="/register"
      >
        <button
          class="auth-button"
        >
          Sign Up/Login
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
